src/ServicePulse.Host/app/js/views/archive/view.html

```diff
@@ -51,7 +51,7 @@
                     </div>
 
                     <div class="col-xs-11 failed-message-data" ng-mouseenter="message.hover2 = true" ng-mouseleave="message.hover2 = false">
-                        <a href="#/failedMessages/{{message.id}}">
+                        <a href="#/message/{{message.id}}">
```

Comment:
Even that it looks nice, I don't like it :D.
With regards to groups, It should be more of:
/group/{{group.id}}/failed-messages
As the group is the parent entity that we care about, and group contains failed messages. So for me ^^ look more natural.
With regards to message path, as we are showing also archived messages I think we should not keep it under failed-message umbrela. Either way, I would tackle this as a separate issue, simply becasue group id path is already out there. And we can do the change of the URL's as a separate initiative with redirects from old paths to new etc.